src/App.tsx

```diff
@@ -1,25 +1,52 @@
-import React from 'react';
+import React, { ChangeEvent, useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { getItemsOnPage } from './utils/pagination/getItemsOnPage';
+import { getItemsOnPageCount } from './utils/pagination/getItemsOnPageCount';
+import { Pagination } from './components/Pagination';
+import { EPaginationSize } from './enums/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState<EPaginationSize>(
+    EPaginationSize.Medium,
+  );
+
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const perPageSizeHandler = (e: ChangeEvent<HTMLSelectElement>) => {
+    const pageSize = +e.target.value;
+
+    if (pageSize === perPage) return;
+
+    setPerPage(pageSize);
+    setCurrentPage(1);
+  };
+
+  const onPageChange = (page: number) => setCurrentPage(+page);
```

Comment:
It's already a number so you don't need unary plus. And this function looks redundant at all